### PR TITLE
FIxed a bug in RollingLife

### PR DESCRIPTION
**Pull request by JulStraus**
When the `RollingLife` lifetime was equal to the duration of a strategic period, we received an error as I forgot to include the index `[t_inv]`. This is now fixed in this PR.

In addition, this PR includes typo corrections.

### CODE REVIEW

File: examples/network.jl [hunk 1]
```diff
@@ -94,15 +94,15 @@ function generate_example_data_network()
             FixedProfile(80),           # Capacity in MW
             FixedProfile(30),           # Variable OPEX in EUR/MWh
             FixedProfile(100),          # Fixed OPEX in EUR/year
-            Dict(NG => 1),              # Output from the Node, in this gase, NG
+            Dict(NG => 1),              # Output from the Node, in this case, NG
             Data[],                         # Potential additional data, no investment for the source
```
Comment (espenfb) [inline examples/network.jl:98]:
```suggestion
            Data[],                     # Potential additional data, no investment for the source
```

File: examples/network.jl [hunk 2]
```diff
@@ -94,15 +94,15 @@ function generate_example_data_network()
...
         ),
         RefSource(                      # Coal source
             "coal source",              # Node id
             FixedProfile(100),          # Capacity in MW
             FixedProfile(9),            # Variable OPEX in EUR/MWh
             FixedProfile(100),          # Fixed OPEX in EUR/year
-            Dict(Coal => 1),            # Output from the Node, in this gase, coal
+            Dict(Coal => 1),            # Output from the Node, in this case, coal
             Data[],                         # Potential additional data, no investment for the source
```
Comment (espenfb) [inline examples/network.jl:106]:
```suggestion
            Data[],                     # Potential additional data, no investment for the source
```

File: test/test_lifetime.jl [hunk 3]
```diff
@@ -2,7 +2,7 @@ resulting_obj = Dict()
 @testset "Test Lifetime" begin
 
     lifetime = 15
-    for sp_dur ∈ [2, 4, 6]#,10,15,20]
+    for sp_dur ∈ [2, 4, 6, 15]#,10,15,20]
```
Comment (espenfb) [inline test/test_lifetime.jl:5]:
```suggestion
    for sp_dur ∈ [2, 4, 6, 15]
```

**Review by espenfb — COMMENTED** [review]
Just some very minor syntax edits/ clean up

**Comment by JulStraus** [comment]
> Just some very minor syntax edits/ clean up

All of them are fine, but I am a bit uncertain whether we have to do the same in the 0.6 branch ;)

**Review by espenfb — APPROVED** [review]

**Pull request merged**